examples/howto_custom_gradients.md

```diff
@@ -0,0 +1,193 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.0
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Use custom gradients
+
+
+JAX provides a convenient `jax.grad` function to evaluate the gradient of any function build with JAX primitives. Which is why Blackjax uses `jax.grad` internally whenever it needs to evaluate the gradient. This should be enough for most applications, but sometimes you may need to provide your own gradients to blackjax for several reasons:
+
+- You have a convenient closed-form expression for the gradient that is evaluated faster than the gradient that JAX produces;
+- The forward-mode differentiation is faster than the backward-mode;
```

Comment:
This is actually the initial motivation of adding this API - could you add a small example of `custom_jvp` with `jax.jacfwd()`?

Reply:
Do you have an example of a logprob where it makes sense to use `jax.jaxfwd()` instead of `jax.grad`?

Reply:
[FedericoV](https://github.com/FedericoV) mentioned a stiff ODE: https://github.com/blackjax-devs/blackjax/issues/201#issuecomment-1255592363

Reply:
I didn't see how to do it immediately; I think the best way is to do something like

```python
import jax.numpy as jnp

def grad(x):
    return jnp.dot(x, jax.jacfwd(logprob_fn)(x))

# Override `jax.grad` in the namespace
```